src/manifest.c

```diff
@@ -878,17 +878,16 @@ void link_submanifests(struct manifest *m1, struct manifest *m2, struct list *su
 			list1 = list1->next;
 			list2 = list2->next;
 
+			/* server (new) manifests will only bring in new bundles */
 			if (file2->last_change > m1->version && !file2->is_deleted) {
-				if (subbed1 && subbed2) {
+				if (!server && subbed1 && subbed2) {
 					account_changed_bundle();
 				} else if (!subbed1 && subbed2) {
 					account_new_bundle();
 				}
 			}
-			if (file2->last_change > m1->version && file2->is_deleted) {
-				if (subbed1) {
+			if (!server && (file2->last_change > m1->version && file2->is_deleted && subbed1)) {
```

Comment:
I liked having the inner conditional check (`if (subbed1) ...`), because it mirrors the conditional block above, making it easier (for me) to understand the code.